src/main/thrift/parquet.thrift

```diff
@@ -288,6 +288,27 @@ struct Statistics {
    8: optional bool is_min_value_exact;
 }
 
+/**
+ * An extension type description
+ *
+ * Extension types allow for third-party semantics not provided by the core
+ * Parquet type system.
+ *
+ * `name` is a dotted name reliably identifying the extension type.
+ * Names beginning with "parquet." are reserved for standardization within
+ * the Parquet project.
+ *
+ * If the extension type is parametric, then `serialization` is an encoding
+ * of the extension type's parameters. It is recommended (but not required)
+ * that the parameters are serialized as a JSON object in UTF-8 encoding.
+ *
+ * If the extension type is not parametric, then `serialization` is empty.
+ */
+struct ExtensionTypeDescription {
```

Comment:
Why choosing a dedicated `ExtensionTypeDescription` struct over `list<KeyValue>`? I'm afraid that a binary typed field may incur misuse from the users.

Reply:
What would the `list<KeyValue>` contain and where would it reside? I'm not following you.

Reply:
```
struct ExtensionTypeDescription {
  1: optional list<KeyValue> metadata
}
```
And specify the required keys for each extension type, pretty much like what Arrow  does.

Reply:
This does not make sense, does it? The keys will always be the same, so why not reify them in the Thrift spec as the PR currently does?

Reply:
Or are you thinking about extension-specific parameter keys as in https://arrow.apache.org/docs/dev/format/CanonicalExtensions.html ?

Reply:
Note we would still need the extension name, so this would be:
```thrift
struct ExtensionTypeDescription {
  1: required string name
  2: optional list<KeyValue> parameters
}
```

Reply:
> Or are you thinking about extension-specific parameter keys as in https://arrow.apache.org/docs/dev/format/CanonicalExtensions.html ?

Yes, I mean something like this.